marconi-chain-index/src/Marconi/ChainIndex/Indexers/Utxo.hs

```diff
@@ -341,13 +387,13 @@ open dbPath (Depth k) isToVacuume = do
                       , value BLOB
                       , inlineScript BLOB
                       , inlineScriptHash BLOB
-                      , slotNo INT NOT NULL
+                      , slotNo NUMERIC NOT NULL
```

Comment:
Isn't numeric worse than int? (as it subject to precision lost)

Reply:
No, it made no difference in my tests.  From the docs, it gets converted to REAL, but still a 64 bit signed.  See [SQLite docs, section 3, Type Affinity](https://www.sqlite.org/datatype3.html)
I also tested it and examined the output using sqlite [setTrace](https://hackage.haskell.org/package/sqlite-simple-0.4.18.2/docs/Database-SQLite-Simple.html#v:setTrace) and still got `-1` for Word64.maxBond.  The relevant code is in the utxoIndexer.hs & commented with TODO.

I have changed back to INT.

So in summary, the fix is store SlotNo as either:
- Text
- Binary
Either way is a trivial change as the  Cardano API has the corresponding instances for SlotNo

Reply:
Yes, but real is a iee float, that is subject to precision loss on high value (the -1 on `maxBound` is puzzling in this case). I think a signed integer is better than a real at the moment, as we should be fine with the 63bits of positive integers for quite a long time). No need to add the confusion of numeric that is a bit further than integer to represent the initial data.